src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -84,24 +86,54 @@ def check_metadata(self):
         validate(metadata_for_validation, schema)
         self.check_extracted_metadata(metadata)
 
-    def check_run_conversion(self, nwbfile_path: str):
+    def check_no_metadata_mutation(self):
+        """Ensure the metadata object was not altered by `add_to_nwbfile` method."""
         metadata = self.interface.get_metadata()
         metadata["NWBFile"].update(session_start_time=datetime.now().astimezone())
 
         metadata_in = deepcopy(metadata)
 
+        nwbfile = mock_NWBFile()
+        self.interface.add_to_nwbfile(nwbfile=nwbfile, metadata=metadata, **self.conversion_options)
+
+        assert metadata == metadata_in
+
+    def check_run_conversion_default_backend(self, nwbfile_path: str, backend: Literal["hdf5", "zarr"] = "hdf5"):
```

Comment:
Whereas for all of these we should check they can write to disk successfully since changing the `DataIO` properties in all kinds of ways can produce errors potentially during `io.write`